components/contentShare/EntityInterface.php

```diff
@@ -0,0 +1,25 @@
+<?php
+
+namespace app\components\contentShare;
+
+interface EntityInterface
+{
+    /**
+     * @return string
+     */
+    public function getContentShareObjectTypeId();
+
+    /**
+     * @return int
+     */
+    public function getContentShareObjectId();
```

Comment:
we have this pattern for a lot of things, including Comments, Stars, Rating. We should probably make a common interface from this.

Reply:
I agree.

I think `objectType` can be done similarly, example:

```php
interface EntityInterface
{
    /**
     * @return string
     */
    public function getObjectType();

    /**
     * @return int
     */
    public function getObjectId();
}

class EntityHelper
{
    const TYPE_NEWS = 'news';
    const TYPE_WIKI = 'wiki';

    public static $typeClasses = [
        self::TYPE_NEWS => News::class,
        self::TYPE_WIKI => Wiki::class,
    ];
}

/**
 * @property int $id
 */
class Wiki extends ActiveRecord implements EntityInterface
{
    public function getObjectType()
    {
        return EntityHelper::TYPE_WIKI;
    }

    public function getObjectId()
    {
        return $this->id;
    }
}

/**
 * @property string $object_type
 * @property int object_id
 */
class Comment extends ActiveRecord implements EntityInterface
{
    public function getObjectType()
    {
        return $this->object_type;
    }

    public function getObjectId()
    {
        return $this->object_id;
    }
}
```

And add trait, example:

Before:
```php
class Comment extends ActiveRecord 
{
    public function getModel()
    {
        if (!in_array($this->object_type, static::$modelClasses, true)) {
            return null;
        }
        /** @var $modelClass ActiveRecord */
        $modelClass = "app\\models\\{$this->object_type}";
        return $modelClass::findOne($this->object_id);
    }
}
```

After:
```php
trait ManagerEntityTrait
{
    public function getModel()
    {
        if (array_key_exists($this->getObjectType(), EntityHelper::$typeClasses)) {
            return EntityHelper::$typeClasses[$this->getObjectType()]::findOne($this->getObjectId());
        }

        throw new InvalidValueException("Class for object type {$this->getObjectType()} not found.");
    }
}

class Comment extends ActiveRecord implements EntityInterface
{
    use ManagerEntityTrait;
}

class Rating extends ActiveRecord implements EntityInterface
{
    use ManagerEntityTrait;
}
```